### PR TITLE
peering: update exported services CRD and bump api module

### PR DESCRIPTION
**Pull request by ndhanushkodi**
Changes proposed in this PR:
- fix the peering acceptance test by updating the exported services CRD to use Peer instead of PeerName to match the config entry
- update the consul api module
- makes a unit test fix with the api module bump-- duplicate server names are no longer allowed across 2 peers

How I've tested this PR:
- unit and acceptance tests

How I expect reviewers to test this PR:
- 👀 

Checklist:
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: control-plane/connect-inject/peering_dialer_controller_test.go [hunk 1]
```diff
@@ -251,7 +251,12 @@ func TestReconcile_CreateUpdatePeeringDialer(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 
 			// Create test consul server.
-			acceptorPeerServer, err := testutil.NewTestServerConfigT(t, nil)
```
Comment (ndhanushkodi) [inline control-plane/connect-inject/peering_dialer_controller_test.go:254]: This change was due to a failure after updating the api module.

Comment (thisisnotashwin) [reply control-plane/connect-inject/peering_dialer_controller_test.go:254]: I think there are a few more tests that need a similar update made to their fixtures.

**Review by jmurret — APPROVED** [review]
👍 thank you for finding and fixing this!

**Review by thisisnotashwin — APPROVED** [review]
Excellent!!!

**Review by t-eckert — APPROVED** [review]
Great find!

**Pull request merged**